### PR TITLE
merge pubsub-hp into master

### PR DESCRIPTION
**Pull request by pongad**

### CODE REVIEW

**Comment by garrettjonesgoogle** [comment]
Take care of any items I mentioned that are trivial to fix, and file issues for ones that are not trivial & fix them after merging.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateSubscriptionAndPullMessages.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016 Google Inc. All Rights Reserved.
```
Comment (pongad) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateSubscriptionAndPullMessages.java]: The file was rewritten, but reverting makes sense.

File: google-cloud-pubsub/pom.xml [hunk 2]
```diff
@@ -54,6 +54,36 @@
       <version>${grpc.version}</version>
     </dependency>
     <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>20.0</version>
```
Comment (pongad) [inline google-cloud-pubsub/pom.xml]: I think this was updated earlier to get stats package which we don't use right now. Reverted.

File: google-cloud-pubsub/pom.xml [hunk 3]
```diff
@@ -54,6 +54,36 @@
...
+    </dependency>
+    <dependency>
+      <groupId>com.google.errorprone</groupId>
+      <artifactId>error_prone_annotations</artifactId>
+      <version>2.0.15</version>
+    </dependency>
+    <dependency>
+      <groupId>joda-time</groupId>
+      <artifactId>joda-time</artifactId>
+      <version>2.9.4</version>
```
Comment (pongad) [inline google-cloud-pubsub/pom.xml]: Done

File: google-cloud-pubsub/pom.xml [hunk 4]
```diff
@@ -54,6 +54,36 @@
...
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-log4j12</artifactId>
+      <version>1.7.21</version>
+    </dependency>
+    <dependency>
+      <groupId>log4j</groupId>
+      <artifactId>log4j</artifactId>
+      <version>1.2.17</version>
+    </dependency>
```
Comment (pongad) [inline google-cloud-pubsub/pom.xml]: Done

File: google-cloud-pubsub/pom.xml [hunk 5]
```diff
@@ -86,6 +122,13 @@
     </profile>
   </profiles>
   <build>
+    <extensions>
+      <extension>
+        <groupId>kr.motd.maven</groupId>
+        <artifactId>os-maven-plugin</artifactId>
+        <version>1.4.0.Final</version>
+      </extension>
+    </extensions>
```
Comment (pongad) [inline google-cloud-pubsub/pom.xml]: It is used to set `os.` properties. Seems to be unused.

File: google-cloud-pubsub/pom.xml [hunk 6]
```diff
@@ -72,6 +102,12 @@
       <version>3.4</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+        <groupId>org.mockito</groupId>
+        <artifactId>mockito-all</artifactId>
+        <version>1.9.5</version>
+        <scope>test</scope>
+    </dependency>
```
Comment (pongad) [inline google-cloud-pubsub/pom.xml]: This is way too big to do right now: https://github.com/GoogleCloudPlatform/google-cloud-java/issues/1596

**Comment by pongad** [comment]
@garrettjonesgoogle PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**